src/Connect.js

```diff
@@ -1,68 +1,450 @@
+                    .then(jwt => this.request(jwt, id))
+  }
+
+  /**
+  *  Create a credential about connnected user
+  *
+  *  @example
+  *  credentials.attest({
+  *   sub: '5A8bRWU3F7j3REx3vkJ...', // uPort address of user, likely a MNID
+  *   exp: <future timestamp>,
+  *   claim: { name: 'John Smith' }
+  *  }).then( credential => {
+  *   ...
+  *  })
+  *
+  * @param    {Object}            [credential]           a unsigned credential object
+  * @param    {String}            credential.claim       claim about subject single key value or key mapping to object with multiple values (ie { address: {street: ..., zip: ..., country: ...}})
+  * @param    {String}            credential.exp         time at which this claim expires and is no longer valid (seconds since epoch)
+  * @param    {String}            [id='attestReq']       string to identify request, later used to get response
+  */
+  attest (claim, id) {
+    this.credentials.attest(claim).then(jwt => {this.request(jwt, id)})
+  }
+
+ /**
+  *  Serializes persistant state of Connect object to string. Persistant state includes following
+  *  keys and values; address, mnid, did, doc, firstReq, keypair. You can save this string how you
+  *  like and then restore it's state with the deserialize function.
+  *
+  *  @return   {String}   JSON string
+  */
+  serialize() {
+    // TODO these are redundant vals, just store did maybe
+    const connectJSONState = {
+      address: this.address,
+      mnid: this.mnid,
+      did: this.did,
+      doc: this.doc,
+      firstReq: this.firstReq,
+      keypair: this.keypair
+    }
+    return JSON.stringify(connectJSONState)
+  }
+
+  /**
+   *  Given string of serialized Connect state, it restores that given state to the Connect
+   *  object which it was called on. You can get the serialized state of a connect object
+   *  by calling the serialize() function.
+   *
+   *  @param    {String}    str      serialized uPort Connect state
+   */
+  deserialize(str) {
+    const state = JSON.parse(str)
+    this.address = state.address
+    this.mnid = state.mnid
+    this.did = state.did
+    this.doc = state.doc
+    this.firstReq = state.firstReq
+    this.keypair = state.keypair
+  }
+
+  /**
+   *  Gets uPort connect state from browser localStorage and sets on this object
+   */
+  getState() {
+    const connectState = store.get('connectState')
+    if (connectState) this.deserialize(connectState)
+  }
+
+    /**
+     *  Writes serialized uPort connect state to browser localStorage at key 'connectState'
+     */
+  setState() {
+    const connectState = this.serialize()
+    store.set('connectState', connectState)
+  }
+
+  /**
+   * Set DID on object, also sets decoded mnid and address
+   */
+  setDID(did) {
+    this.did = did
+    this.mnid = did.replace('did:ethr:', '').replace('did:uport:', '')
+    this.address = decode(this.mnid).address
```

Comment:
@zachferland  This fails when `did === 'did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74'`